### DISCUSSION TITLE
The Policies formatter updates the policy to something invalid that stops returning queries

### DISCUSSION THREAD

**Discussion by saikatdas0790** [supabase/supabase#3960] [Questions]
My policy looks like this

```
(organization_member_of = (select organization_member_of from user_profile where id = auth.uid()))
```

which is formatted to this

```
(organization_member_of = ( SELECT user_profile_1.organization_member_of FROM user_profile user_profile_1 WHERE (user_profile_1.id = uid())))
```

The data in my table looks like this

![image](https://user-images.githubusercontent.com/7412443/142645275-3b69268d-c909-438c-83c3-dd91eea417d7.png)

If I run a SQL query like this, it returns all 3 rows. Here I am hardcoding the `uid` of the first user to simulate the RLS situation. Which leads me to believe that my RLS policy is correct.

![image](https://user-images.githubusercontent.com/7412443/142645545-47407446-69e0-4af3-a079-8a9e9045856f.png)

Am I missing something or does the policy editor have a bug?

**Comment by bnjmnt4n** [comment]
The formatting changes to the policy editor are expected, and done automatically by Postgres when saving the policy. If there are any syntax errors with the policy SQL, Postgres will instead fail to add the policy. Instead you might need to add separate policies to the `user_profile` table as well, since you are referencing that table in this policy. (Running SQL queries on the SQL editor bypasses any RLS policies.)

**Reply by saikatdas0790** [reply]
The policy that I'm applying here is on the user_profile table only.

Basically the policy that I'm setting here is allow selects on user_profile to a logged in member for only the entries where their org id matches the org id. So, that the logged in user can only query data for the same org that they belong to and not others
